chatkit-core/src/main/kotlin/com/pusher/chatkit/cursors/CursorSubscriptionEventParser.kt

```diff
@@ -31,7 +31,7 @@ internal object CursorSubscriptionEventParser : DataParser<CursorSubscriptionEve
 
     private fun JsonObject.parseEvent(eventName: String): Result<CursorSubscriptionEvent, Error> =
             when (eventName) {
-                "new_cursor" -> parseAs<Cursor>().map(CursorSubscriptionEvent::OnCursorSet)
+                "new_cursor" -> parseAs<CursorSubscriptionEvent.OnCursorSet>()
```

Comment:
I don't think the current version will work, shouldn't it be:
```suggestion
                "new_cursor" -> parseAs<CursorApiType>().map(CursorSubscriptionEvent::OnCursorSet)
```
?

Reply:
Fair point! Have updated this part!